bitcoin/tx.c

```diff
@@ -398,6 +398,7 @@ struct bitcoin_tx *bitcoin_tx(const tal_t *ctx,
 	tx->input_amounts = tal_arrz(tx, struct amount_sat*, input_count);
 	tx->wtx->locktime = 0;
 	tx->wtx->version = 2;
+	tx->output_witscripts = tal_arrz(tx, struct witscript*, output_count);
```

Comment:
We usually don't zero our allocations right away (prefer `tal_arr`) since that'd mask accesses to uninitialized memory that can be caught with `valgrind`. In this case it's ok since you want to have a NULL-pointer out of the gate.

Reply:
Yes, want them NULL in the cases where we don't set a witscript.